### PR TITLE
FAT-821: Implement e-2-e automation of test case C3501: An item is being moved from one library location to another. Update the effective location for the item.

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
Refs: https://issues.folio.org/browse/FAT-821

### CODE REVIEW

**Review by TetianaParanich — APPROVED** [review]

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import ItemRecordView from '../../support/fragments/inventory/itemRecordView';
+import TopMenu from '../../support/fragments/topMenu';
+import getRandomPostfix from '../../support/utils/stringTools';
+
+describe('Update the effective location for the item', () => {
+  const instanceTitle = `autoTestInstanceTitle.${getRandomPostfix()}`;
+  const anotherPermanentLocation = 'Main Library';
+  const ITEM_BARCODE = Number(new Date()).toString();
+
+  before(() => {
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:15]: use before each instead

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    cy.log(ITEM_BARCODE);
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:16]: do you need such cy.log()?

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    cy.login(Cypress.env('diku_login'), Cypress.env('diku_password'));
+    cy.getToken(Cypress.env('diku_login'), Cypress.env('diku_password'))
+      .then(() => {
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:19]: please check, but ".then(() =>{" looks like excess in such cases

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import getRandomPostfix from '../../support/utils/stringTools';
+
+describe('Update the effective location for the item', () => {
+  const instanceTitle = `autoTestInstanceTitle.${getRandomPostfix()}`;
+  const anotherPermanentLocation = 'Main Library';
+  const ITEM_BARCODE = Number(new Date()).toString();
+
+  before(() => {
+    cy.log(ITEM_BARCODE);
+    cy.login(Cypress.env('diku_login'), Cypress.env('diku_password'));
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:17]: "Logged into FOLIO as user with Cataloging permission set"

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import InventoryInstance from '../../support/fragments/inventory/inventoryInstance';
+import InventorySearch from '../../support/fragments/inventory/inventorySearch';
+import ItemRecordView from '../../support/fragments/inventory/itemRecordView';
+import TopMenu from '../../support/fragments/topMenu';
+import getRandomPostfix from '../../support/utils/stringTools';
+
+describe('Update the effective location for the item', () => {
+  const instanceTitle = `autoTestInstanceTitle.${getRandomPostfix()}`;
+  const anotherPermanentLocation = 'Main Library';
+  const ITEM_BARCODE = Number(new Date()).toString();
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:13]: could you move item_barcode generation to common helper/related fragment?

File: cypress/integration/inventory/update-effective-location-for-item.spec.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  afterEach(() => {
+    cy.getInstances({ limit: 1, expandAll: true, query: `"items.barcode"=="${ITEM_BARCODE}"` })
+      .then(() => {
+        cy.deleteItem(Cypress.env('instances')[0].items[0].id);
+        cy.deleteHoldingRecord(Cypress.env('instances')[0].holdings[0].id);
+        cy.deleteInstanceApi(Cypress.env('instances')[0].id);
+      });
+  });
+
+  it('C3501 Update the effective location for the item', { tags: [TestTypes.smoke] }, () => {
```
Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:61]: sync naming in code and testrail. I see "An item is being moved from one library location to another. Update the effective location for the item." in testrail now

Comment (dmitriyfolio) [inline cypress/integration/inventory/update-effective-location-for-item.spec.js:61]: I haven't found implementation of verifications:
"The changes are saved and you are taken back to the detailed holdings view."
" The effective location is displayed in the top of the Item record"

**Review by dmitriyfolio — APPROVED** [review]

**Pull request merged**